PDSim/scroll/core.py

```diff
@@ -2011,34 +1990,102 @@ def lump_energy_balance_callback(self):
+            print('    Torque (pV Analysis):', self.tau_mechanical,'Nm')
+            print('    Mechanical losses:', self.mech.Wdot_losses,'kW')
+            print('    Motor losses:', self.motor.losses,'kW')
             print('    Electrical power:', self.Wdot_electrical,'kW')
             print('    Mass flow rate:', self.mdot,'kg/s')
             if hasattr(self,'Wdot_i'):
                 print('    Over. isentropic:', self.eta_oi,'-')
             if hasattr(self,'eta_v'):
                 print('    Volumetric:', self.eta_v,'-')
         
-        #Want to return a list
-        return [Qnet]
+        if not hasattr(self,'OEB_type') or self.OEB_type == 'single-lump':
+    
+            #For the single lump
+            # HT terms are positive if heat transfer is TO the lump
+            Qnet = 0.0
+            Qnet -= sum([Tube.Q for Tube in self.Tubes])
+            
+            self.Qamb = self.ambient_heat_transfer(self.Tlumps[0])
+            
+            self.mech.Wdot_losses = self.mechanical_losses('low:shell') 
+            # Shaft power from forces on the orbiting scroll from the gas in the pockets [kW]
+            self.Wdot_forces = self.omega*self.forces.mean_tau
+    #        if self.geo.is_symmetric():
+    #            
+    #        else:
+    #            import warnings
+    #            warnings.warn('No ML for asymmetric for now')
+    #            self.mech.Wdot_losses = 0.0
+            
+            # Heat transfer with the ambient; Qamb is positive if heat is being removed, thus flip the sign
+            Qnet -= self.Qamb
+            
+            Qnet += self.mech.Wdot_losses
+            # Heat transfer with the gas in the working chambers.  mean_Q is positive
+            # if heat is transfered to the gas in the working chamber, so flip the 
+            # sign for the lump
+            Qnet -= self.HTProcessed.mean_Q       
+        
+            return [Qnet]
+        
+        elif self.OEB_type == 'multi-lump':
+            """
+            Example with two lumps
+            Tlumps[0] : Tshell
+            Tlumps[1] : Toil
+            """
+            #For the lumps:
+            Qnet = 0.0
+            Qnet_oil = 0.0
+            
+            # HT terms are positive if heat transfer is TO the lump
+            Qnet -= sum([Tube.Q for Tube in self.Tubes])
+            
+            # HT Shell to ambient
+            self.Qamb = self.ambient_heat_transfer(self.Tlumps[0])
+            
+            # Heat transfer with the ambient; Qamb is positive if heat is being removed, thus flip the sign
+            Qnet -= self.Qamb
+            
+            # Heat transfer with the gas in the working chambers.  mean_Q is positive
+            # if heat is transfered to the gas in the working chamber, so flip the 
+            # sign for the lump
+            Qnet -= self.HTProcessed.mean_Q
+            
+            #HT oil shell
+            self.Rshell_oil = 190 #K/kW  from Chen (2000) - PhD thesis
```

Comment:
This definitely doesn't feel like it should be hardcoded in core, and rather should call out to a function that can be overwritten in subclass.

Reply:
I can give it some more thoughts about the best way to do it and then push the changes. Ideally, we would like it to be as general as possible.